power-management/rapl/powercap/runtest.sh

```diff
@@ -0,0 +1,201 @@
+    typeset -i i
+    for (( i = 0; i < $nthreads; i++ )); do
+        (load1 $timeout) > /dev/null 2>&1 &
+    done
+}
+
+# === power-management/rapl/powercap/test_capping.sh ==========================
+function test_capping
+{
+    # measurement time should be bigger than
+    # cat /sys/devices/virtual/powercap/intel-rapl/intel-rapl\:0/constraint_0_time_window_us
+    # which is 1 sec by default (at least on systems I have seen so far).
+    typeset measurement_time=30
+    typeset load_time=$(( measurement_time + 20 ))
+
+    # 1. no load, no cap
+    typeset e_no_no=$(get_energy $measurement_time)
+    rlLog "Power consumtion on uncapped system with no load: $e_no_no microwatts"
+
+    # 2. load, no cap
+    typeset cpus=$(lscpu | grep '^CPU(s):' | \
+                   sed 's/^CPU(s):[^0-9]*\([0-9]*\)/\1/')
+    loadm $load_time $cpus
+    typeset e_full_no=$(get_energy $measurement_time)
+    rlLog "Power consumtion on uncapped system with full load: $e_full_no microwatts"
+
+    # 3. cap to average between full load and no load
+    typeset e_mid=$(( (e_no_no + e_full_no) / 2 ))
+    rlLog "Going to cap the system to: $e_mid microwatts"
+
+    typeset old_cap=$(get_cap)
+    set_cap $e_mid
+    loadm $load_time $cpus
+    typeset e_full_cap=$(get_energy $measurement_time)
+    rlLog "Power consumtion on capped system with full load: $e_full_cap microwatts"
+
+    #
+    # measured_max is maximal measured value which will not cause fail
+    # measured_min is minimal measured value which will not cause fail
+    # *11/10 - add 10% in bash which works in integers only - we're
+    # using microwatts, so numbers are milions to hundreds of milions,
+    # so tehre is no problem with rounding error
+    # *9/10 - substract 10%
+    #
+    typeset measured_max=$(( e_mid * 11 / 10 ))
+    typeset measured_min=$(( e_mid *  9 / 10 ))
+
+    rlLog "capping back to $old_cap"
+    set_cap $old_cap
+
+    if (( e_full_cap > measured_max || e_full_cap < measured_min )); then
+        rlLog "FAIL: measured energy consumption doesn't match capped value"
+        return 1
+    else
+        rlLog "PASS: Test passed"
+    fi
+
+    # All my tests can write warnings (like non met depedencies) to this
+    # file, which wil be translated to warning in the Beaker.
+    touch warn.tmp
+}
+
+# === power-management/rapl/powercap/warn.sh ==================================
+function powercap_warn
+{
+    typeset warn_file="./warn.tmp"
+    rlRun -l "cat $warn_file"
+
+    typeset lines=$(wc -l $warn_file | sed 's/^\([0-9]*\).*/\1/')
+    rlRun -l "rm -f $warn_file" "0-255"
+
+    rlPhaseStart WARN ""
+        rlAssertEquals "Warnings occured during test." "$lines" 0
+    rlPhaseEnd
+}
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "TmpDir=\$(mktemp -d)" 0 "Creating tmp directory"
+    rlPhaseEnd
+
+    rlPhaseStartTest
+        #debug
+        rlRun -l "find /sys/devices/ -name *rapl*"
+        rlRun -l "lsmod"
+        rlRun -l "modprobe intel-rapl"
```

Comment:
Does this test apply only to Intel CPUs? If so, should we check the vendor and SKIP if it's not applicable instead of failing the test when Intel-specific module fails to load?